### PR TITLE
Update CI workflows

### LINKED ISSUE #1067: Building on CI fails due to missing packages
Hi @imagejan @nicokiaru @ctrueden,

When building on the GitHub CI it throws these errors:

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project mobie-viewer-fiji: Compilation failure: Compilation failure: 
Error:  /home/runner/work/mobie-viewer-fiji/mobie-viewer-fiji/src/main/java/org/embl/mobie/lib/MoBIEHelper.java:[48,21] package spimdata.util does not exist
Error:  /home/runner/work/mobie-viewer-fiji/mobie-viewer-fiji/src/main/java/org/embl/mobie/lib/hcs/OperettaMetadata.java:[3,35] package ch.epfl.biop.bdv.img.opener does not exist
Error:  /home/runner/work/mobie-viewer-fiji/mobie-viewer-fiji/src/main/java/org/embl/mobie/lib/SpimDataAdder.java:[45,21] package spimdata.util does not exist
Error:  /home/runner/work/mobie-viewer-fiji/mobie-viewer-fiji/src/main/java/org/embl/mobie/DataStore.java:[34,38] package ch.epfl.biop.bdv.img.imageplus does not exist
Error:  /home/runner/work/mobie-viewer-fiji/mobie-viewer-fiji/src/main/java/org/embl/mobie/lib/hcs/Plate.java:[49,46] package ch.epfl.biop.bdv.img.bioformats.entity does not exist
```

However, when building locally with `mvn` and also when running it in Fiji everything seems fine...could it be the issue that those are transitive dependencies that come from [mobie-io](https://github.com/mobie/mobie-io/blob/main/pom.xml)?

### PR DESCRIPTION
**Pull request by imagejan**
I ran `scijava-scripts/github-actionify.sh -f` locally, and removed the `deploy.yml` and `test.yml` files in favor of the newly created `build.yml`.

This hopefully closes #1067.

### CODE REVIEW

**Comment by tischi** [comment]
...and in fact the tests are running now: https://github.com/mobie/mobie-viewer-fiji/actions/runs/7113664024

I think the issue might have been that the latest mobie-io was not yet on maven or something similar.

**Comment by imagejan** [comment]
Alright, then I'd suggest keeping the steps for blosc and xvfb, but always using `.github/build.sh`. Is there any need to keep `test` and `deploy` separate? The `ci-build.sh` script pulled in from `scijava-scripts` already has logic to decide between SNAPSHOT and release builds and deploys them accordingly.

**Comment by tischi** [comment]
@imagejan all right I will try to convert back and merge your PR.

I think essentially I have to add in a few lines of code into the `build.yml`.

I wonder why we had this:

```
      - name: Cache m2 folder
        uses: actions/cache@v2
        env:
          cache-name: cache-m2
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-build-${{ env.cache-name }}
          restore-keys: |
            ${{ runner.os }}-build-${{ env.cache-name }}-
            ${{ runner.os }}-build-
            ${{ runner.os }}-

      - name: Set up JDK 8
        uses: actions/setup-java@v2
        with:
          java-version: '8'
          distribution: 'zulu'
```

Do you know? Maybe that was something that used to be there in the old times?

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,48 +0,0 @@
...
-        uses: actions/setup-java@v2
-        with:
-          java-version: '8'
-          distribution: 'zulu'
-      - name: Set up CI environment
-        run: .github/setup.sh
-      - name: Install blosc
-        run: sudo apt install libblosc-dev
-      - name: Install xvfb
-        run: sudo apt-get install xvfb
```
Comment (imagejan) [inline .github/workflows/deploy.yml:39]: We should then probably migrate these lines into the new `build.yml`.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,45 +0,0 @@
-name: test
-
-on:
-  pull_request:
-    branches:
-      - main
-      - develop
```
Comment (imagejan) [inline .github/workflows/test.yml:7]: @tischi  do you follow a dev pattern where PRs to `develop` are common? Then `- develop` should also be kept in the `build.yml` file added in this PR.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://github.com/mobie/mobie-viewer-fiji/actions/workflows/build.yml/badge.svg)](https://github.com/mobie/mobie-viewer-fiji/actions/workflows/build.yml)
+
 [![](https://github.com/mobie/mobie-viewer-fiji/actions/workflows/build-main.yml/badge.svg)](https://github.com/mobie/mobie-viewer-fiji/actions/workflows/build-main.yml)
```
Comment (imagejan) [inline README.md:3]: I forgot to remove old badges. This one here was already invalid before this PR, so let's remove it.

```suggestion
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'zulu'
+          cache: 'maven'
+      - name: Set up CI environment
+        run: .github/setup.sh
+      - name: Execute the build
+        run: .github/build.sh
```
Comment (imagejan) [inline .github/workflows/build.yml:28]: So let's add blosc and xvfb back:

```suggestion
      - name: Set up CI environment
        run: .github/setup.sh
      - name: Install blosc
        run: sudo apt install libblosc-dev
      - name: Install xvfb
        run: sudo apt-get install xvfb
      - name: Execute the build
        run: .github/build.sh
```

**Review by imagejan — COMMENTED** [review]
I think the current CI setup is a mix of previous versions of the SciJava-based setup, with some custom requirements added over time.
@tischi if you agree with my comments in general, I can update this PR accordingly in the next few days.

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (tischi) [reply .github/workflows/deploy.yml:39]: Yes.

File: .github/workflows/test.yml (hunk 2 above)
Comment (tischi) [reply .github/workflows/test.yml:7]: Yes.

File: README.md (hunk 3 above)
Comment (tischi) [reply README.md:3]: Thanks!

**Comment by tischi** [comment]
Thanks a lot @imagejan, everything you did and said looks very good! 
Happy to merge this once you give the go.

**Comment by imagejan** [comment]
@tischi I commit the pending changes and rebased this pull request onto the `main` branch. Should be good to merge now.

**Pull request merged**